integration/integration_test.go

```diff
@@ -0,0 +1,454 @@
+		cmd.run(t)
+	}
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	if !*integration {
+		fmt.Println("Skipping integration test.")
+
+		os.Exit(0)
+	}
+
+	if os.Getenv("KUBECTL_GADGET") == "" {
+		fmt.Fprintf(os.Stderr, "please set $KUBECTL_GADGET.")
+
+		os.Exit(-1)
+	}
+
+	if *image != "" {
+		os.Setenv("GADGET_IMAGE_FLAG", "--image "+*image)
+	}
+
+	initCommands := []*command{
+		deployInspektorGadget,
+		waitUntilInspektorGadgetPodsDeployed,
+		waitUntilInspektorGadgetPodsInitialized,
+		createTestNamespace,
+	}
+
+	fmt.Printf("Setup inspektor-gadget:\n")
+	for _, cmd := range initCommands {
+		err := cmd.runWithoutTest()
+
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+
+			os.Exit(-1)
+		}
+	}
+
+	ret := m.Run()
+
+	fmt.Printf("Clean inspektor-gadget:\n")
+	cleanupTestNamespace.runWithoutTest()
+	cleanupInspektorGadget.runWithoutTest()
+
+	os.Exit(ret)
+}
+
+func TestBindsnoop(t *testing.T) {
+	bindsnoopCmd := &command{
+		name:           "Start bindsnoop gadget",
+		cmd:            "$KUBECTL_GADGET bindsnoop -n test-ns",
+		expectedRegexp: `test-ns\s+test-pod\s+test-pod\s+\d+\s+nc`,
+		startAndStop:   true,
+	}
+
+	commands := []*command{
+		bindsnoopCmd, // Start it.
+		{
+			name:           "Run pod which calls bind()",
+			cmd:            "kubectl run --restart=Never --image=busybox -n test-ns test-pod -- sh -c 'while true; do nc -l -p 9090 -w 1; done'",
+			expectedString: "pod/test-pod created\n",
+		},
+		waitUntilTestPodReady,
+		{
+			name: "Wait so bind() are called",
+			cmd:  "sleep 15",
+		},
+		deleteTestPod,
+		bindsnoopCmd, // Stop it.
+	}
+
+	runCommands(commands, t)
+}
+
+func TestBiolatency(t *testing.T) {
+	commands := []*command{
+		{
+			name:           "Run biolatency gadget",
+			cmd:            "id=$($KUBECTL_GADGET biolatency start --node $(kubectl get node --no-headers | cut -d' ' -f1)); sleep 15; $KUBECTL_GADGET biolatency stop $id",
+			expectedRegexp: `usecs\s+:\s+count\s+distribution`,
+		},
+	}
+
+	runCommands(commands, t)
+}
+
+func TestBiotop(t *testing.T) {
+	biotopCmd := &command{
+		name:           "Start biotop gadget",
+		cmd:            "$KUBECTL_GADGET biotop --node $(kubectl get node --no-headers | cut -d' ' -f1)",
+		expectedRegexp: `etcd`,
+		startAndStop:   true,
+	}
+
+	commands := []*command{
+		biotopCmd, // Start it.
+		{
+			name: "Wait a bit.",
+			cmd:  "sleep 10",
+		},
+		biotopCmd, // Stop it.
+	}
+
+	runCommands(commands, t)
+}
+
+func TestCapabilities(t *testing.T) {
+	capabilitiesCmd := &command{
+		name:           "Start capabilities gadget",
+		cmd:            "$KUBECTL_GADGET capabilities -n test-ns",
+		expectedRegexp: `test-ns\s+test-pod.*nice.*CAP_SYS_NICE`,
+		startAndStop:   true,
+	}
+
+	commands := []*command{
+		capabilitiesCmd, // Start it.
+		{
+			name:           "Run pod which fails to run nice",
+			cmd:            "kubectl run --restart=Never --image=busybox -n test-ns test-pod -- sh -c 'while true; do nice -n -20 echo; done'",
```

Comment:
Nit: You can create a `runBusyboxPod` method to avoid repeating this so many times:

```
fun RunBusyboxPod(cmd string) string {
  return fmt.Sprintf(""kubectl run --restart=Never --image=busybox -n test-ns test-pod -- sh -c '%s'",
 cmd)
}
```

And this line will become:

```
cmd:            runBusyboxPod("while true; do nice -n -20 echo; done"),
```